modules/diskq/qdisk.c

```diff
@@ -1173,6 +1173,11 @@ qdisk_start(QDisk *self, const gchar *filename, GQueue *qout, GQueue *qbacklog,
 
   if (new_file)
     {
+      if (self->options->prealloc)
+        {
+          _truncate_file(self, qdisk_get_maximum_size(self));
```

Comment:
The `ftruncate` does not pre allocate space, or it does not have to. It can actually just pretend that there is a file that size and actually can later a write can fail due to not enough space on the disk. (sparse file). There is a `posix_fallocate` method that with a correct arguments actually does write the zeros to the disk. (which probably much slower)

A short demo based on this gist: https://gist.github.com/jpauli/8afec7c4fc2b38f8ff27

```bash
/tmp $ df -h /tmp/
Filesystem      Size  Used Avail Use% Mounted on
tmpfs            16G  388M   16G   3% /tmp
/tmp $ ./gistfile1 1 1073741824 #1gb
now opening /tmp/__allocate_chunk
Using ftruncate
Done !, st_blocks=0
/tmp $ df -h /tmp/
Filesystem      Size  Used Avail Use% Mounted on
tmpfs            16G  388M   16G   3% /tmp
/tmp $ ls -hali /tmp/__allocate_chunk
24116 -rw-r--r-- 1 kokan kokan 1.0G Jul  1 11:33 /tmp/__allocate_chunk
/tmp $ rm /tmp/__allocate_chunk
/tmp $ ./gistfile1 2 1073741824 #1gb, fallocate
now opening /tmp/__allocate_chunk
Using fallocate
Done !, st_blocks=2097152
/tmp $ df -h /tmp/
Filesystem      Size  Used Avail Use% Mounted on
tmpfs            16G  1.4G   15G   9% /tmp
/tmp $ ls -hali /tmp/__allocate_chunk
24117 -rw-r--r-- 1 kokan kokan 1.0G Jul  1 11:34 /tmp/__allocate_chunk
/tmp $
```

Reply:
Thanks, fixed!